### PR TITLE
Fix default return value for node when using submit button

### PR DESCRIPTION
**Pull request by akchinSTC**
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: elyra/pipeline/parser.py [hunk 1]
```diff
@@ -143,7 +143,8 @@ def _create_pipeline_operation(node: Dict, super_node: Optional[Dict] = None):
             id=node_id,
             type=node.get('type'),
             classifier=node.get('op'),
-            name=PipelineParser._get_ui_data_field(node, 'label'),
+            name=PipelineParser._get_ui_data_field(node, 'label',
+                                                   default_value=PipelineParser._get_app_data_field(node, 'filename')),
```
Comment (lresende) [inline elyra/pipeline/parser.py:147]: you have filename a few lines below, don't want to move this down below and use that as the default value?

**Review by ptitzler — APPROVED** [review]
Submit now seems to work for Python scripts, R scripts and notebooks.

There's still the issue with the bogus file extension for notebooks but it doesn't seem to have an impact on this issue.

image

File: elyra/pipeline/parser.py (hunk 1 above)
Comment (kevin-bates) [reply elyra/pipeline/parser.py:147]: I agree.  I guess I was surprised to see the default value added here, but then I noticed that the Operation constructor requires a value for `name` and that the defaulting performed within the `name` property accessor is really all about dropping the suffix.  So, this default value seems fine.

Regarding leveraging the `filename` keyword value as suggested, I'm not sure if that would work.  Can a keyword reference another keyword in a call? No - just checked.  You'll get `NameError: name 'filename' is not defined`.  You'd need to move it into a regular variable prior to the instance construction.

**Review by kevin-bates — APPROVED** [review]
Since the issue is resolved, I'm cool with dealing with the default value detail until later.

**Review by lresende — APPROVED** [review]

**Pull request merged**